src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/KestrelHttpParser.cs

```diff
@@ -109,6 +109,11 @@ public KestrelHttpParser(IKestrelTrace log)
 
                     pathEncoded = true;
                 }
+                else if (ch == 0)
```

Comment:
Why is this being checked here now? Should blow up when trying to get ASCII string.